core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -63,6 +67,28 @@ private void preparePartitionedTable() {
         .commit();
   }
 
+  @Test
+  public void testManifestsTableWithDroppedPartition() throws IOException {
+    table.newFastAppend()
+        .appendFile(FILE_A)
```

Comment:
I think for completeness sake we should also have tests for the other Partition reading metadata tables. (Files, Manfiests, ManifestEntries) or what not

Reply:
Should the following tests(`testAllManifestsTableWithDroppedPartition` and `testFilesTableScanWithDroppedPartition`) have covered the case ?

Reply:
ManfiestEntries and partitions should be slightly different right?

Reply:
Added